peps/pep-0748.rst

```diff
@@ -0,0 +1,735 @@
+for it. Nonetheless, third-party libraries make use of some of its undocumented
+attributes. For instance, `pydantic <https://github.com/pydantic/pydantic/blob/00ff77ed37589d924d3c10e0d5a48a7ef679a0d7/pydantic/v1/typing.py#L66>`__
+and `typeguard <https://github.com/agronholm/typeguard/blob/016f8139f5a0a63147d68df9558cc5584cd2c49a/src/typeguard/_utils.py#L44>`__
+use the ``_evaluate`` method; `beartype <https://github.com/beartype/beartype/blob/0b4453f83c7ed4be054d8733aab8075e1478e166/beartype/_util/hint/pep/proposal/pep484585/utilpep484585ref.py#L210>`__
+and `pyanalyze <https://github.com/quora/pyanalyze/blob/9e401724f9d035cf138b72612834b6d5a00eb8e8/pyanalyze/annotations.py#L509>`__
+use the ``__forward_arg__`` attribute.
+
+We replace the existing but poorly specified :py:class:`!typing.ForwardRef` with a new class,
+``annotations.ForwardRef``. It is designed to be mostly compatible with existing uses
+of the :py:class:`!typing.ForwardRef` class, but without the behaviors specific to the
+:py:mod:`!typing` module. For compatibility with existing users, we keep the private
+``_evaluate`` method, but mark it as deprecated. It delegates to a new public function in
+the :py:mod:`!typing` module, ``typing.evaluate_forward_ref``, that is designed to
+evaluate forward references in a way that is specific to type hints.
+
+We add a function ``annotations.call_annotate_function`` as a helper for calling
+``__annotate__`` functions. This is a useful building block when implementing functionality
+that needs to partially evaluate annotations while a class is being constructed.
+For example, the implementation of :py:class:`typing.NamedTuple` needs to retrieve
+the annotations from a class namespace dictionary before the namedtuple class itself
+can be constructed, because the annotations determine what fields exist on the namedtuple.
+
+Specification
+-------------
+
+A new module, ``annotations``, is added to the standard library. Its aim is to
+provide tooling for introspecting and wrapping annotations.
+
+The exact contents of the module are not yet specified. We will add support for
+:pep:`649` semantics to standard library functionality that uses annotations, such
+as :py:mod:`dataclasses` and :py:class:`typing.TypedDict`, and use the experience
+to inform the design of the new module.
+
+The module will contain the following functionality:
+
+* ``get_annotations()``: A function that returns the annotations of a function,
+  module, or class. This will replace :py:func:`inspect.get_annotations`. The latter
+  will delegate to the new function. It may eventually be deprecated, but to
+  minimize disruption, we do not propose an immediate deprecation.
+* ``Format``: an enum that contains the possible formats of annotations. This will
+  replace the ``VALUE``, ``FORWARDREF``, and ``SOURCE`` formats in :pep:`649`.
+  :pep:`649` proposed to make these values global members of the :py:mod:`inspect`
```

Comment:
No need to link the second mention in this paragraph, especially as it's immediately after the first:
```suggestion
  replace the ``VALUE``, ``FORWARDREF``, and ``SOURCE`` formats in :pep:`649`.
  PEP 649 proposed to make these values global members of the :py:mod:`inspect`
```